Userland/Utilities/wasm.cpp

```diff
@@ -38,6 +39,201 @@ static void sigint_handler(int)
     g_continue = false;
 }
 
+static Optional<u128> convert_to_uint(StringView string)
+{
+    if (string.is_empty())
+        return {};
+
+    u128 value = 0;
+    auto const characters = string.characters_without_null_termination();
+
+    for (size_t i = 0; i < string.length(); i++) {
+        if (characters[i] < '0' || characters[i] > '9')
+            return {};
+
+        value *= 10;
+        value += u128 { static_cast<u64>(characters[i] - '0'), 0 };
+    }
+    return value;
+}
+
+static Optional<u128> convert_to_uint_from_hex(StringView string)
+{
+    if (string.is_empty())
+        return {};
+
+    u128 value = 0;
+    auto const count = string.length();
+    auto const upper_bound = NumericLimits<u128>::max();
+
+    for (size_t i = 0; i < count; i++) {
+        char digit = string[i];
+        u8 digit_val;
+        if (value > (upper_bound >> 4))
+            return {};
+
+        if (digit >= '0' && digit <= '9') {
+            digit_val = digit - '0';
+        } else if (digit >= 'a' && digit <= 'f') {
+            digit_val = 10 + (digit - 'a');
+        } else if (digit >= 'A' && digit <= 'F') {
+            digit_val = 10 + (digit - 'A');
+        } else {
+            return {};
+        }
+
+        value = (value << 4) + digit_val;
+    }
```

Comment:
This all reads very similar to `AK::decode_hex(_digit)`, can't we reuse any of that?

Reply:
None of the hex readers I looked at could handle u128, I'll take a look at that one too.

Reply:
Oh that decided into an allocated buffer, producing a u128 from that is probably more code tbh

Reply:
Maybe just decode_hex_digit then?

Reply:
Sure thing.